BnbnavNetClient/Services/NetworkOperations/RoadUpdateOperation.cs

```diff
@@ -0,0 +1,58 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Avalonia.Media;
+using BnbnavNetClient.Models;
+using BnbnavNetClient.Views;
+
+namespace BnbnavNetClient.Services.NetworkOperations;
+
+public class RoadUpdateOperation : NetworkOperation
+{
+    private readonly MapEditorService _editorService;
+    private readonly Road _road;
+    private readonly string _newName;
+    private readonly RoadType _newRoadType;
+    private readonly List<Edge> _affectedEdges = new();
+
+    public RoadUpdateOperation(MapEditorService editorService, Road road, string newName, RoadType newRoadType)
+    {
+        _editorService = editorService;
+        _road = road;
+        _newName = newName;
+        _newRoadType = newRoadType;
+        
+        _affectedEdges.AddRange(editorService.MapService!.Edges.Values.Where(x => x.Road == road));
+    }
+    
+    public override async Task PerformOperation()
+    {
+        ItemsNotToRender.AddRange(_affectedEdges);
+        
+        try
+        {
+            (await _editorService.MapService!.Submit($"/roads/{_road.Id}", new
+            {
+                name = _newName,
+                type = _newRoadType.ServerName()
+            })).AssertSuccess();
```

Comment:
consider setting to a variable

Reply:
There's nothing to set to a variable here apart from the response which I think should be inline if we are only ensuring that it is successful

Reply:
Resolving now - if we want to action this change we should apply it to all operations simultaneously in another PR